src/lib/annotations/AnnotationDialog.js

```diff
@@ -388,41 +403,43 @@ const CLASS_ANIMATE_DIALOG = 'bp-animate-show-dialog';
 
         switch (dataType) {
             // Clicking 'Post' button to create an annotation
-            case 'post-annotation-btn':
+            case constants.CLASS_ANNOTATION_BUTTON_POST:
                 this.postAnnotation();
                 break;
             // Clicking 'Cancel' button to cancel the annotation
-            case 'cancel-annotation-btn':
+            case constants.CLASS_ANNOTATION_BUTTON_CANCEL:
                 if (this.isMobile) {
+                    // Hide mobile dialog without destroying the thread
                     this.hideMobileDialog();
                 } else {
+                    // Cancels + destroys the annotation thread
                     this.cancelAnnotation();
                 }
 
                 this.deactivateReply(true);
                 break;
             // Clicking inside reply text area
-            case 'reply-textarea':
+            case CLASS_REPLY_TEXTAREA:
                 this.activateReply();
                 break;
             // Canceling a reply
-            case 'cancel-reply-btn':
+            case DATA_TYPE_CANCEL_REPLY:
                 this.deactivateReply(true);
                 break;
             // Clicking 'Post' button to create a reply annotation
-            case 'post-reply-btn':
+            case DATA_TYPE_POST_REPLY:
                 this.postReply();
                 break;
             // Clicking trash icon to initiate deletion
-            case 'delete-btn':
+            case DATA_TYPE_DELETE:
                 this.showDeleteConfirmation(annotationID);
                 break;
             // Clicking 'Cancel' button to cancel deletion
-            case 'cancel-delete-btn':
+            case CLASS_CANCEL_DELETE:
```

Comment:
I'm a tiny bit hesitant for us to use the same variable, for a class, as a data-type property. Can you make constants for the data-types, and use those instead?